core/src/ast/modules.rs

```diff
@@ -34,6 +34,17 @@ impl Module {
             .for_each(|m| m.check_opaque(&in_path.sub_path(self.name.clone()), env, errors));
     }
 
+    /// Ensures that we are not exporting any non-opaque zero-sized types
+    pub fn check_zst<'a>(&'a self, in_path: &Path, errors: &mut Vec<Path>) {
```

Comment:
If we keep adding more checks, it might be nice to consolidate the checks to use a visitor instead someday. But right now the checks operate on different node types so it's probably not worth the effort.

Reply:
Yeah, I'd also like to be able to test them